addons/explore-editor-theme/ui/IconSaver.gd

```diff
@@ -0,0 +1,34 @@
+tool
+extends VBoxContainer
+
+
+signal filesystem_changed()
+
+export var icon_name := ''
+export var type_name := ''
+
+onready var _save_dialog: FileDialog = $SaveDialog
```

Comment:
I don't really prefix node references with `_` for some reason, but I guess there is no harm in that. They are not expected to be accessed from the outside anyway.